src/main/java/com/example/accommodationbookingapp/repository/booking/BookingRepository.java

```diff
@@ -22,4 +24,9 @@ public interface BookingRepository extends JpaRepository<Booking, Long> {
 
     @EntityGraph(attributePaths = {"accommodation", "user"})
     void deleteByIdAndUserId(Long bookingId, Long userId);
+
+    @Query("SELECT b FROM Booking b JOIN FETCH b.accommodation a LEFT JOIN FETCH a.amenities"
+            + " WHERE b.user.id = :userId AND b.checkInDate BETWEEN :startDate AND :endDate")
```

Comment:
I suggest using Java Text Block here to avoid concatenation
```suggestion
    @Query("""
        SELECT b
        FROM Booking b
        JOIN FETCH b.accommodation a
        LEFT JOIN FETCH a.amenities
        WHERE b.user.id = :userId
        AND b.checkInDate BETWEEN :startDate AND :endDate
        """)
```